src/main.rs

```diff
@@ -44,37 +47,57 @@ impl Direction {
 }
 
 impl Game {
-    pub fn new() -> Game {
+    pub fn new(start_level: u16) -> Game {
         // Get input ready
         let input = input();
         input
             .disable_mouse_mode()
             .expect("Can't disable mouse mode");
 
-        let mut initial_snake_parts = HashSet::<(u16, u16)>::new();
-        initial_snake_parts.insert((0, 0));
-        initial_snake_parts.insert((1, 0));
-        initial_snake_parts.insert((2, 0));
+        let terminal_width: u16 = get_terminal_size().0;
 
+        let mut initial_snake_parts = HashSet::<(u16, u16)>::new();
         let mut initial_ordered_snake_parts = VecDeque::<(u16, u16)>::new();
-        initial_ordered_snake_parts.push_back((0, 0));
-        initial_ordered_snake_parts.push_back((1, 0));
-        initial_ordered_snake_parts.push_back((2, 0));
 
-        Game {
+        //create snake with size of starting level (+3 because you start with a length of 3)
+        //if the snake is longer than the terminal width it loops on the line below so it doesn't crash
+        //for example with a terminal width of 4 and a length of 15
+        //_____________
+        //| 1| 2| 3| 4|
+        //| 6| 7| 8| 5|
+        //|11|12| 9|10|
+        //|  |13|14|15|
+        //-------------
+        for i in 1..(start_level + 3) {
+            initial_snake_parts.insert(((i - (i / terminal_width)) % terminal_width, i / terminal_width));
+        }
+
+        for i in 1..(start_level + 3) {
+            initial_ordered_snake_parts.push_back(((i - (i / terminal_width)) % terminal_width, i / terminal_width));
+        }
+
+        let mut new_game: Game = Game {
             snake: Snake {
-                direction: Direction::Right,
+                //if there is less than 5 cells before the snake will crash into itself then set the direction to down
+                direction: if start_level % terminal_width >= terminal_width - 5 { Direction::Down } else { Direction::Right },
```

Comment:
Is this right? I think this works for the first line (like level 30), but as the snake loops around the map, this doesn't really help much at later levels like 60. Maybe a better technique would be to always start going down, since there's almost always much space downwards?